Makefile

```diff
@@ -0,0 +1,125 @@
+################################################################################
+# Variables
+################################################################################
+
+ERTS_INCLUDE_DIR ?= $(shell erl -noshell -s init stop -eval "io:format(\"~s/erts-~s/include/\", [code:root_dir(), erlang:system_info(version)]).")
+
+CFLAGS := -g -std=c11 -Wall -Wextra -Wshadow -Wfloat-equal -Wundef -Wcast-align \
+	-Wwrite-strings -Wunreachable-code -Wformat=2 -Wswitch-enum \
+	-Wswitch-default -Winit-self -Wno-strict-aliasing -I$(ERTS_INCLUDE_DIR)
```

Comment:
I think you disabled the strict aliasing warnings but did not disable strict aliasing itself. You can do that with `-fno-strict-aliasing`